src/App.test.js

```diff
@@ -1,16 +1,46 @@
-import { render, screen } from '@testing-library/react';
+import { render, screen, waitFor } from '@testing-library/react';
 import App from './App';
 import React from 'react'
+import * as fetchJson from "./fetchJson";
+import AstronautsTable from './AstronautsTable';
 
 
-test('renders learn react link', () => {
+test('renders loading screen', () => {
   render(<App />);
-  const linkElement = screen.getByText(/learn react/i);
-  expect(linkElement).toBeInTheDocument();
+  const loadingText = screen.getByText(/Loading/i);
+  expect(loadingText).toBeInTheDocument();
 });
 
-test('renders test message', () => {
+test('renders graphic', () => {
   render(<App />);
-  const linkElement = screen.getByText(/Grace's app/i);
-  expect(linkElement).toBeInTheDocument();
+  const isGraphicThere = document.getElementsByClassName("Astronaut");
+  expect(isGraphicThere.length).toBe(1);
+});
+
+test('renders graphic', () => {
+  render(<App />);
+  const isTableThere = document.getElementsByTagName("header");
+  expect(isTableThere.length).toBe(1);
+});
+
+test('fetchJson gets called', () => {
+  const spy = jest.spyOn(fetchJson, 'default');
+  render(<App />);
+  expect(spy).toHaveBeenCalled();
+});
+
+test('data from fetchJson is displayed', async () => {
+  const json = "{\"people\": [{\"craft\": \"Rocketship\", \"name\": \"David Bowie\"}], \"message\": \"success\", \"number\": 1}"
+  const jsonJson = JSON.parse(json);
```

Comment:
Er, `jsonJson` is a bit confusing. Maybe make the first variable `jsonString` and this variable `jsonObject`?

Reply:
that makes a lot of sense!